homeassistant/config_entries.py

```diff
@@ -911,6 +912,9 @@ def _abort_if_unique_id_configured(
                         self.hass.async_create_task(
                             self.hass.config_entries.async_reload(entry.entry_id)
                         )
+                if entry.source == SOURCE_IGNORE and remove_if_ignored:
+                    asyncio.create_task(entry.async_remove(self.hass))
```

Comment:
It should not be removed at this point in time. There is no new entry yet.
The actual removal should take place when the new entry is created.

Reply:
diving in the code - IIUC this is already happening: https://github.com/home-assistant/core/blob/40e5634db3366bc5e37cb9c0a4735c76cb8e7421/homeassistant/config_entries.py#L512